src/library/scala/concurrent/BatchingExecutor.scala

```diff
@@ -91,7 +91,7 @@ private[concurrent] trait BatchingExecutor extends Executor {
       // if we know there will be blocking, we don't want to keep tasks queued up because it could deadlock.
       {
         val tasks = _tasksLocal.get
-        _tasksLocal set Nil
+        _tasksLocal set null
```

Comment:
https://github.com/scala/scala/pull/10528/files#diff-6506ab3dbea2fcdfe9ed21a3e513b1387540dcd2396871472ce283be42dbda2bR54-R55

Reply:
Yes, exactly. And if it's `Nil` - it will always throw exception.